src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -282,6 +283,44 @@ public CreateFluidSourceEvent(World world, BlockPos pos, IBlockState state)
             super(world, pos, state);
         }
     }
+
+    /**
+     * Fired when liquids mix to create a new block.
+     */
+    public static class FluidMixEvent extends BlockEvent
+    {
+        private final Material liquid1;
+        private final Material liquid2;
+        private IBlockState block;
+
+        public FluidMixEvent(World world, BlockPos pos, Material liquid1, Material liquid2, IBlockState block)
```

Comment:
This seems to only be fired for `WATER` and `LAVA`. Are there other cases where fluids mix? Can two fluids with material `WATER` ever mix and fire an event?

Also it might be better to pass the fluid block positions instead of materials so that someone can check the properties of their fluid and create more advanced mixes.

Reply:
Not in vanilla.

Minecraft doesn't necessarily track the positions by default, and it is possible that lava could mix with water coming from two directions. Perhaps if I provided the position of the block that triggered the event alongside the position of the block to be placed. A mod author can use those if they need to check for more complicated interactions.

By that logic the materials of the liquids in question should be pretty easy to determine. I think removing those and adding a second BlockPos for the block that triggered the event is the way to go.